File: public/usage-examples/graphics/text_width_font_named-1-example.cpp
#include "splashkit.h"

int main()
{
    open_window("Text Width", 680, 200);
    clear_screen(COLOR_WHITE);

    load_font("LOTR", "lotr.ttf");
    set_font_style("LOTR", BOLD_FONT);
    draw_text("Ringbearer", COLOR_BLACK, "LOTR", 100, 30, 20);

    //Getting the width of the text to fill a rectange of that width
    int width = text_width("Ringbearer", "LOTR", 100);
    fill_rectangle(COLOR_BLACK, 30, 130, width, 10);
    refresh_screen();

    delay(5000);
    close_all_windows();

    return 0;
}

File: public/usage-examples/graphics/text_width_font_named-1-example.py
from splashkit import *

open_window("Text Width", 680, 200)
clear_screen(color_white())

load_font("LOTR", "lotr.ttf")
set_font_style_name_as_string("LOTR", FontStyle.bold_font)
draw_text_font_as_string("Ringbearer", color_black(), "LOTR", 100, 30, 20)

#Getting the width of the text to fill a rectange of that width
width = text_width_font_named("Ringbearer", "LOTR", 100)
fill_rectangle(color_black(), 30, 130, width, 10)
refresh_screen()

delay(5000)
close_all_windows()